### PR TITLE
[HttpKernel] Deprecate bundle inheritance

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #24048
| License       | MIT
| Doc PR        | symfony/symfony-docs#8389

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 1]
```diff
@@ -206,6 +206,10 @@ public function getBundles()
      */
     public function getBundle($name, $first = true)
     {
+        if (false === $first) {
+            @trigger_error(sprintf('Passing "false" as the second argument to %s() is deprecated as of 3.4 and will be removed in 4.0.', __METHOD_), E_USER_DEPRECATED);
```
Comment (pierredup) [inline src/Symfony/Component/HttpKernel/Kernel.php]: Typo, __METHOD_ should be __METHOD__

Comment (fabpot) [reply src/Symfony/Component/HttpKernel/Kernel.php]: fixed

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 2]
```diff
@@ -206,6 +206,10 @@ public function getBundles()
      */
     public function getBundle($name, $first = true)
     {
+        if (false === $first) {
```
Comment (ro0NL) [inline src/Symfony/Component/HttpKernel/Kernel.php]: at least `locateResource` is calling it with false.. should we get rid of $first there too? I believe it serves the same purpose.

Comment (fabpot) [reply src/Symfony/Component/HttpKernel/Kernel.php]: Correct, fixed

**Review by ro0NL — APPROVED** [review]

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 3]
```diff
@@ -204,8 +204,12 @@ public function getBundles()
     /**
      * {@inheritdoc}
      */
-    public function getBundle($name, $first = true)
+    public function getBundle($name, $first = true, $noDeprecation = false)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Kernel.php]: signature change should be removed, `func_get_arg` should be used instead.

File: src/Symfony/Component/HttpKernel/Bundle/BundleInterface.php [hunk 4]
```diff
@@ -56,6 +56,8 @@ public function getContainerExtension();
      * bundle.
      *
      * @return string The Bundle name it overrides or null if no parent
+     *
+     * @deprecated This method is deprecated as of 3.4 and will be removed in 4.0.
```
Comment (xabbuh) [inline src/Symfony/Component/HttpKernel/Bundle/BundleInterface.php]: Do we need to drop the final dot here according to the latest CS changes?

**Comment by chalasr** [comment]
Should we also deprecate `Bundle::getParent()` default implem before removing it?

File: src/Symfony/Component/HttpKernel/Bundle/BundleInterface.php (hunk 4 above)
Comment (fabpot) [reply src/Symfony/Component/HttpKernel/Bundle/BundleInterface.php]: Most @deprecated tags have a dot at the end currently. As this is a sentence, I think that's the way it should be.

File: src/Symfony/Component/HttpKernel/Kernel.php (hunk 3 above)
Comment (fabpot) [reply src/Symfony/Component/HttpKernel/Kernel.php]: fixed

**Comment by fabpot** [comment]
@chalasr Not sure it makes sense. This method is mainly called by Symfony itself, and if a bundle defines a parent, the message won't be displayed. So, it seems to me better to not add this.

**Comment by fabpot** [comment]
RFR

**Review by chalasr — APPROVED** [review]

**Pull request merged**